### PR TITLE
Apply elm-format to generated elm files

### PR DESCRIPTION
**Pull request by martimatix**
This is a PR to apply elm format to the generated files. Happy to make any changes or answer any questions.

### CODE REVIEW

File: src/graphqelm.ts [hunk 1]
```diff
@@ -35,14 +36,40 @@ if (!graphqlUrl) {
...
+    '--stdin',
+    '--output',
+    path
+  ])
+
+  elmFormat.stdin.write(value)
+  elmFormat.stdin.end()
+
+  elmFormat.stdout.on('data', data => {
+    console.log(data.toString())
```
Comment (martimatix) [inline src/graphqelm.ts]: This should never run and is only here for completeness.

File: src/graphqelm.ts [hunk 2]
```diff
@@ -35,14 +36,40 @@ if (!graphqlUrl) {
...
+  })
+
+  elmFormat.stderr.on('data', data => {
+    console.log(data.toString())
```
Comment (martimatix) [inline src/graphqelm.ts]: If elm-format cannot auto format the file - the error message will be output to the terminal.

File: src/graphqelm.ts [hunk 3]
```diff
@@ -35,14 +36,40 @@ if (!graphqlUrl) {
...
+  })
+
+  elmFormat.on('close', code => {
+    if (code !== 0) {
+      console.log(`elm-format process exited with code ${code}`)
```
Comment (martimatix) [inline src/graphqelm.ts]: This should never is only here for completeness.

**Comment by dillonkearns** [comment]
Hey Mario, thanks so much for the PR! This is awesome. I did run into an issue when running this from a directory that doesn't have a `node_modules` folder in it (for example, if I run it from within the `examples` folder, which has `../node_modules`). That made me think that some users will also have `elm-format` installed globally, while others will want to use it from within a `node_modules` folder somewhere (perhaps in a directory below them as in my case above).

I can think of a few ways to handle this:

1) Have some sort of flag to let the user indicate the `elm-format` executable path to use, and skip formatting otherwise (this is the more conservative approach). For example, `graphqelm --elm-format=../node_modules/.bin/elm-format`.
2) Search down a few directories for a `node_modules` folder (here's a popular npm package that does this: https://www.npmjs.com/package/find-root). It's actually pretty common for elm projects to not have a package.json file I've noticed. So I guess this could fall back to looking for a global installation, or just skip it if none is found?

Really nice approach with the `writeWithElmFormat` function, that's really readable and declarative! I'm open to other approaches if you can think of any.

**Comment by martimatix** [comment]
Thanks for the feedback, @dillonkearns. I pushed a change that references elm-format relative to the absolute path of the node file that is being executed. In this case, that would be `dist/bundle.js`. What do you think?

File: webpack.config.ts [hunk 4]
```diff
@@ -4,6 +4,9 @@ const path = require('path')
 module.exports = {
   entry: './src/graphqelm.ts',
   target: 'node',
+  node: {
+    __dirname: false
```
Comment (martimatix) [inline webpack.config.ts]: Without this, the value of `__dirname` is just `'/'`. It's counter-intuitive that setting it to `false` injects the value. [Reference](https://github.com/webpack/webpack/issues/1599#issuecomment-186841345)

**Comment by dillonkearns** [comment]
Looks great! Users shouldn't really need to look at the source files, but if they did for some reason want to use their own version of `elm-format` they could just run it manually after running `graphqelm`. Seems like a simple and reliable approach 👍 I'll just upgrade to 0.7.0 (the latest experimental version of `elm-format`) after I merge this in. It's just as stable as 0.6.1.

**Pull request merged**